### PR TITLE
Add additional logging to track SLA violation

### PR DESCRIPTION
**Pull request by djaiswal83**
Add additional logging to track why once in a while flow level SLA is not honored.

### CODE REVIEW

File: azkaban-common/src/main/java/azkaban/trigger/builtin/SlaChecker.java [hunk 1]
```diff
@@ -69,16 +69,23 @@ public static SlaChecker createFromJson(final HashMap<String, Object> obj)
...
+      logger.info("Start time is less than 0 for flow " + flow.toString());
       return false;
     }
 
     Status status;
     if (type.getComponent() == SlaType.ComponentType.FLOW) {
+      logger.info("SLA type is flow.");
       if (this.checkTime < flow.getStartTime()) {
+        logger.info("checktime = " + this.checkTime);
+        logger.info("SLA duration = " + slaOption.getDuration().toMillis());
```
Comment (jamiesjc) [inline azkaban-common/src/main/java/azkaban/trigger/builtin/SlaChecker.java]: Consider adding the unit for the logging, e.g. `ms`?

File: azkaban-common/src/main/java/azkaban/trigger/builtin/SlaChecker.java [hunk 2]
```diff
@@ -69,16 +69,23 @@ public static SlaChecker createFromJson(final HashMap<String, Object> obj)
...
         this.checkTime = flow.getStartTime() + slaOption.getDuration().toMillis();
+        logger.info("checktime updated to " + this.checkTime);
       }
       status = flow.getStatus();
+      logger.info("Flow status = " + status);
```
Comment (jamiesjc) [inline azkaban-common/src/main/java/azkaban/trigger/builtin/SlaChecker.java]: Would `status.toString()` be better?

File: azkaban-common/src/main/java/azkaban/trigger/builtin/SlaChecker.java [hunk 3]
```diff
@@ -92,15 +99,18 @@ private Boolean isSlaMissed(final ExecutableFlow flow) {
...
          case FLOW_SUCCEED:
+           logger.info("isFlowSucceeded?");
             return !isFlowSucceeded(status);
          case JOB_FINISH:
             return !isJobFinished(status);
          case JOB_SUCCEED:
             return !isJobFinished(status);
       }
     } else if (slaOption.getType().getStatus() == StatusType.SUCCEED) {
+      logger.info("slaOption.status = SUCCEED and status = " + status);
```
Comment (jamiesjc) [inline azkaban-common/src/main/java/azkaban/trigger/builtin/SlaChecker.java]: Same ^^ `status.toString()`

File: azkaban-common/src/main/java/azkaban/trigger/builtin/SlaChecker.java [hunk 4]
```diff
@@ -162,6 +172,7 @@ public Object isSlaFailed() {
     final ExecutableFlow flow;
     try {
       flow = this.executorLoader.fetchExecutableFlow(this.execId);
+      logger.info("Flow for execid " + this.execId + " is " + flow.toString());
```
Comment (jamiesjc) [inline azkaban-common/src/main/java/azkaban/trigger/builtin/SlaChecker.java]: Why to you need to print out `flow.toString()`? Would `flow.getId()` be better?

File: azkaban-common/src/main/java/azkaban/trigger/builtin/SlaChecker.java (hunk 1 above)
Comment (djaiswal83) [reply azkaban-common/src/main/java/azkaban/trigger/builtin/SlaChecker.java]: Looking at line 84, I believed checkTime would be in ms only.

File: azkaban-common/src/main/java/azkaban/trigger/builtin/SlaChecker.java (hunk 2 above)
Comment (djaiswal83) [reply azkaban-common/src/main/java/azkaban/trigger/builtin/SlaChecker.java]: Makes sense, I will do that.

File: azkaban-common/src/main/java/azkaban/trigger/builtin/SlaChecker.java (hunk 3 above)
Comment (djaiswal83) [reply azkaban-common/src/main/java/azkaban/trigger/builtin/SlaChecker.java]: ditto

File: azkaban-common/src/main/java/azkaban/trigger/builtin/SlaChecker.java (hunk 4 above)
Comment (djaiswal83) [reply azkaban-common/src/main/java/azkaban/trigger/builtin/SlaChecker.java]: agreed

**Pull request merged**